backup: TestAlterBackupScheduleWithSQLSpecialCharacters failed

backup.TestAlterBackupScheduleWithSQLSpecialCharacters [failed](https://mesolite.cluster.engflow.com/invocations/default/78b3baba-d17e-41bb-9678-78d32b070656?testReportRun=1&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [2b7c2f9cae67ac95520159d81edf44194657bac8](https://github.com/cockroachdb/cockroach/commits/2b7c2f9cae67ac95520159d81edf44194657bac8):

Fatal error:

```
panic: failed to load dry-run backup job: failed to get progress for job 1155524442049544193: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/54/1/1155524442049544193/"legacy_progress"/1913-10-29T12:58:44.828991999Z/0): "sql txn" meta={id=00d69644 key=/Tenant/10/Table/15/1/1155524442049544193/0 iso=Serializable pri=0.01963601 epo=0 ts=1772708474.496061538,2 min=1772708474.171008017,0 seq=0} lock=true stat=PENDING rts=1772708474.171008017,0 gul=1772708474.671008017,0 obs={n1@1772708474.171008017,0}

```
Stack: 

```
goroutine 4103 [running]:
github.com/cockroachdb/cockroach/pkg/util/besteffort.Warning({0xc323d70, 0xc00e439540}, {0xac7cbaa, 0x14}, 0xc0050df1e0)
	pkg/util/besteffort/besteffort.go:34 +0x267
github.com/cockroachdb/cockroach/pkg/backup.invokeBackup({_, _}, _, _, {_, _})
	pkg/backup/schedule_exec.go:175 +0x27c
github.com/cockroachdb/cockroach/pkg/backup.dryRunInvokeBackup({_, _}, {_, _}, _)
	pkg/backup/create_scheduled_backup.go:579 +0x20a
github.com/cockroachdb/cockroach/pkg/backup.dryRunBackup({_, _}, {_, _}, _)
	pkg/backup/create_scheduled_backup.go:565 +0x125
github.com/cockroachdb/cockroach/pkg/backup.doAlterBackupSchedules({0xc323d70, 0xc00e439540}, {0xc4102e8, 0xc00429cd40}, 0xc003d7d5e0, 0xc003b1f810)
	pkg/backup/alter_backup_schedule.go:185 +0xac5
github.com/cockroachdb/cockroach/pkg/backup.alterBackupScheduleHook.func1({0xc323d70, 0xc00e439540}, 0xc003b1f810)
	pkg/backup/alter_backup_schedule.go:794 +0x69
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0xc323d70, 0xc00e439540})
	pkg/sql/planhook.go:194 +0x13c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xc323d70, 0xc00e439540})
	pkg/util/stop/stopper.go:440 +0x112
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 4025
	pkg/util/stop/stopper.go:438 +0x247
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAlterBackupScheduleWithSQLSpecialCharacters
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAlterBackupScheduleWithSQLSpecialCharacters2576604735
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterBackupScheduleWithSQLSpecialCharacters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61080

--- a/pkg/backup/schedule_exec.go
+++ b/pkg/backup/schedule_exec.go
@@ -176,6 +176,12 @@ func invokeBackup(
 		select {
 		case res := <-resultCh:
 			event, err = getBackupFnTelemetry(ctx, registry, txn, res)
+			if sql.ErrIsRetryable(err) {
+				log.Dev.Warningf(ctx,
+					"best effort operation 'get-backup-telemetry' failed with retryable error: %+v", err,
+				)
+				return nil
+			}
 			return err
 		case <-ctx.Done():
 			return ctx.Err()

